### ISSUE TITLE
Insights

### ISSUE THREAD

**Issue by psaegert** [MrWhatZitToYaa/IGNNS-final-project#15]
- Dropout in Nested Neural Network does not seem to work very well
- Dropout in Feature Network works well
- Too few conditions (~16) are not enough, even if they are outputs if a feature network. ~1000 (perhaps a little too much) work well
- Dropout after GELU works better than before
- large batch sizes lead to instability (?)
- dropout in feature network is important
- ActNorm seems to help

Notable Runs:
```python
# Training
N=1000
batch_size=256
lr=1e-4
patience=100
factor=0.8

# CNF
hidden_size=128
n_blocks=32
act_norm=True

# Feature Network
hidden_sizes=[1024] * 5
condition_size = 1024
dropout=0.5
```
![image](https://github.com/MrWhatZitToYaa/IGNNS-final-project/assets/36567814/a04f7c0c-db26-4442-a21c-530406e4df35)

```python
# Training
N=2000
batch_size=256
lr=1e-4
patience=200
factor=0.8

# CNF
hidden_size=128
n_blocks=32
act_norm=True

# Feature Network
hidden_sizes=[1024] * 5
condition_size = 1024
dropout=0.5
```
![image](https://github.com/MrWhatZitToYaa/IGNNS-final-project/assets/36567814/2bcf5552-ebb1-41a0-a6b2-f216c8d9526c)
![image](https://github.com/MrWhatZitToYaa/IGNNS-final-project/assets/36567814/d9f455ff-1d8f-4292-ac7e-9ce716e53932)
